mitiq/cdr/cdr.py

```diff
@@ -120,6 +121,12 @@ def execute_with_cdr(
                 "Must provide `num_fit_parameters` for custom fit function."
             )
 
+    # Check if circuit is already Clifford
+    if is_clifford(circuit):
+        if not isinstance(simulator, Executor):
+            simulator = Executor(simulator)
```

Comment:
Suggestion: to avoid code duplication you can remove
```
if not isinstance(simulator, Executor):
            simulator = Executor(simulator)
```
and just place, before `# Check if circuit is already Clifford` , the existing initialization block that you can find later in the code, i.e.:

```
if not isinstance(executor, Executor):
    executor = Executor(executor)
if not isinstance(simulator, Executor):
    simulator = Executor(simulator)
```

Reply:
Yes that's better, thanks!